dex/elrond_dex_router/src/util.rs

```diff
@@ -0,0 +1,90 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+#[elrond_wasm_derive::module]
+pub trait UtilModule {
+    #[inline]
+    fn is_active(&self) -> bool {
+        self.state().get()
+    }
+
+    #[endpoint(setPairCreationEnabled)]
+    fn set_pair_creation_enabled(&self, enabled: bool) -> SCResult<()> {
+        self.require_owner()?;
+        self.pair_creation_enabled().set(&enabled);
+        Ok(())
+    }
+
+    fn require_owner(&self) -> SCResult<()> {
+        let caller = self.blockchain().get_caller();
+        let owner = self.owner().get();
+        require!(caller == owner, "Permission denied");
+        Ok(())
+    }
+
+    fn send_tokens(
+        &self,
+        token: &TokenIdentifier,
+        amount: &Self::BigUint,
+        destination: &Address,
+        opt_accept_funds_func: &OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        if amount > &0 {
+            let (function, gas_limit) = match opt_accept_funds_func {
+                OptionalArg::Some(accept_funds_func) => (
+                    accept_funds_func.as_slice(),
+                    self.transfer_exec_gas_limit().get(),
+                ),
+                OptionalArg::None => {
+                    let no_func: &[u8] = &[];
+                    (no_func, 0u64)
+                }
+            };
+
+            let result = self.send().direct_esdt_execute(
+                destination,
+                token,
+                amount,
+                gas_limit,
+                function,
+                &ArgBuffer::new(),
+            );
+
+            match result {
```

Comment:
I think you can use "?" operator on `Result<T>`.

Reply:
done